Sources/TABResourceLoader/Protocols/ResourceOperationType.swift

```diff
@@ -1,58 +0,0 @@
-//
-//  ResourceOperationType.swift
-//  TABResourceLoader
-//
-//  Created by Luciano Marisi on 10/09/2016.
-//  Copyright © 2016 Luciano Marisi. All rights reserved.
-//
-
-import Foundation
-
-/// Define a type that is cancellable
-public protocol Cancellable: class {
-  /// Returns whether or not the type has been cancelled
-  var cancelled: Bool { get }
-}
-
-public protocol Finishable: class {
-  /**
-   Method to be called when the type finished all it's work
-   */
-  func finish()
-}
-
-public protocol ResourceOperationType: Cancellable, Finishable {
-
-  associatedtype ResourceService: ResourceServiceType
-
-  /**
-   Fetches a resource using the provided service
-
-   - parameter resource: The resource to fetch
-   - parameter service:  The service to be used for fetching the resource
-   */
-  func fetch(resource: ResourceService.Resource, usingService service: ResourceService)
```

Comment:
the implementation of this function move to a private function called on `execute`, so I have not moved the docs